pkg/autoscaler/autoscaler/tidb_autoscaler.go

```diff
@@ -59,8 +60,28 @@ func syncTiDBAfterCalculated(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbCluster
 	intervalSeconds := tac.Spec.TiDB.ScaleInIntervalSeconds
 	if recommendedReplicas > currentReplicas {
 		intervalSeconds = tac.Spec.TiDB.ScaleOutIntervalSeconds
+		if tac.Status.TiDB.Phase != v1alpha1.ReadyToScaleOut {
+			tac.Status.TiDB.Phase = v1alpha1.ReadyToScaleOut
+			// phase could change from Normal to ReadyToScaleOut, ReadyToScaleIn to ReadyToScaleOut,
+			// reset timestamp in both cases.
+			tac.Annotations[label.AnnTiDBReadyToScaleTimestamp] = fmt.Sprintf("%d", time.Now().Unix())
+		}
+	} else {
+		if tac.Status.TiDB.Phase != v1alpha1.ReadyToScaleIn {
+			tac.Status.TiDB.Phase = v1alpha1.ReadyToScaleIn
+			// phase could change from Normal to ReadyToScaleIn, ReadyToScaleOut to ReadyToScaleIn,
+			// reset timestamp in both cases.
+			tac.Annotations[label.AnnTiDBReadyToScaleTimestamp] = fmt.Sprintf("%d", time.Now().Unix())
+		}
+	}
+	ableToScale, err := checkStsReadyAutoScalingTimestamp(tac, 123)
+	if err != nil {
+		return err
+	}
+	if !ableToScale {
+		return nil
 	}
-	ableToScale, err := checkStsAutoScalingInterval(tac, *intervalSeconds, v1alpha1.TiDBMemberType)
+	ableToScale, err = checkStsAutoScalingInterval(tac, *intervalSeconds, v1alpha1.TiDBMemberType)
```

Comment:
I think we could extract `checkStsReadyAutoScalingTimestamp` and `checkStsAutoScalingInterval` into one function as `checkStsAutoScaling`, then we would call `checkStsReadyAutoScalingTimestamp` and `checkStsAutoScalingInterval` in this function.